beacon-chain/sync/rpc_beacon_blocks_by_range.go

```diff
@@ -214,6 +217,37 @@ func (s *Service) validateRangeRequest(r *pb.BeaconBlocksByRangeRequest) error {
 	return nil
 }
 
+// filters all the provided blocks to ensure they are canonical
+// and are strictly linear.
+func (s *Service) filterBlocks(ctx context.Context, blks []*ethpb.SignedBeaconBlock, roots [][32]byte, prevRoot *[32]byte,
+	step, startSlot uint64) ([]*ethpb.SignedBeaconBlock, error) {
+
+	newBlks := make([]*ethpb.SignedBeaconBlock, 0, len(blks))
+
+	for i, b := range blks {
+		isRequestedSlotStep := (b.Block.Slot-startSlot)%step == 0
```

Comment:
Should we validate start slot <= b.Block.Slot ?

Reply:
all this validation is already done and taken care of in `validateRangeRequest`